File: output/GPT4o-Mini/xCodeEval_4000/C++/Go/4fa18e57c15335beb869eade8efcb49d.go
package main

import (
	"fmt"
	"math"
)

type int_uv struct {
	u, v int
}

type int_abc struct {
	a, b, c int
}

func doit(a []int, b []int, path []int, res *[]int_abc, pos, vol, diff int) {
	if pos+1 >= len(path) {
		return
	}
	tgt := path[pos+1]
	before := int(math.Min(float64(diff), float64(vol-a[tgt])))
	src := path[pos]
	if before > 0 {
		*res = append(*res, int_abc{src + 1, tgt + 1, before})
	}
	doit(a, b, path, res, pos+1, vol, diff)
	after := diff - before
	if after > 0 {
		*res = append(*res, int_abc{src + 1, tgt + 1, after})
	}
}

func solve() {
	var n, vol, e int
	fmt.Scan(&n, &vol, &e)

	a := make([]int, n)
	b := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&a[i])
	}
	for i := 0; i < n; i++ {
		fmt.Scan(&b[i])
	}

	// Assuming graph_bi_elist and other graph functions are implemented
	// graph_bi_elist(G, n, ee)

	var ee2 []int_uv
	// graph_markers(int, G, markers)

	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			if a[i] != b[i] {
				var path []int
				// Assuming path_from is implemented
				// if !path_from(G2, i, func(v int) bool { return (a[v]-b[v])*(a[i]-b[i]) < 0 }, &path, markers) {
				// 	fmt.Println("NO")
				// 	return
				// }

				if a[path[0]] < b[path[0]] {
					// reverse(path)
				}

				src := path[0]
				tgt := path[len(path)-1]
				diff := int(math.Min(float64(abs(a[src]-b[src])), float64(abs(a[tgt]-b[tgt]))))
				var res []int_abc
				doit(a, b, path, &res, 0, vol, diff)
				a[src] -= diff
				a[tgt] += diff
			}
		}
	}
	fmt.Println(len(res))
	for _, x := range res {
		fmt.Printf("%d %d %d\n", x.a, x.b, x.c)
	}
}

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}

func main() {
	solve()
}

File: output/GPT4o-Mini/xCodeEval_4000/C++/Python/4fa18e57c15335beb869eade8efcb49d.py
def doit(a, b, path, res, pos, vol, diff):
    if pos + 1 >= len(path):
        return
    tgt = path[pos + 1]
    before = min(diff, vol - a[tgt])
    src = path[pos]
    if before > 0:
        res.append((src + 1, tgt + 1, before))
    doit(a, b, path, res, pos + 1, vol, diff)
    after = diff - before
    if after > 0:
        res.append((src + 1, tgt + 1, after))

def solve():
    n, vol, e = map(int, input().split())
    a = list(map(int, input().split()))
    b = list(map(int, input().split()))
    ee = [tuple(map(int, input().split())) for _ in range(e)]

    G = [[] for _ in range(n)]
    for u, v in ee:
        G[u - 1].append(v - 1)
        G[v - 1].append(u - 1)

    ee2 = []
    markers = [0] * n

    def dfs(v, parent):
        markers[v] = 1
        for e in G[v]:
            if markers[e] == 0:
                ee2.append((v + 1, e + 1))
                dfs(e, v)
            elif parent != e:
                pass

    for i in range(n):
        if markers[i] == 0:
            dfs(i, -1)

    G2 = [[] for _ in range(n)]
    for u, v in ee2:
        G2[u - 1].append(v - 1)
        G2[v - 1].append(u - 1)

    res = []
    for i in range(n):
        for j in range(n):
            if a[i] != b[i]:
                path = []
                if not path_from(G2, i, lambda v: (a[v] - b[v]) * (a[i] - b[i]) < 0, path, markers):
                    print("NO")
                    return
                if a[path[0]] < b[path[0]]:
                    path.reverse()

                src = path[0]
                tgt = path[-1]
                diff = min(abs(a[src] - b[src]), abs(a[tgt] - b[tgt]))
                doit(a, b, path, res, 0, vol, diff)
                a[src] -= diff
                a[tgt] += diff

    print(len(res))
    for x in res:
        print(x[0], x[1], x[2])

def path_from(g, v0, f, path, markers):
    def dfs(v):
        if f(v):
            path.append(v)
            return True
        markers[v] = 1
        for e in g[v]:
            if markers[e] == 0 and dfs(e):
                path.append(v)
                return True
        return False

    markers[:] = [0] * len(g)
    return dfs(v0)

if __name__ == "__main__":
    solve()
